### ISSUE TITLE
Fatal Python Error: deallocating None. (Version 5.2.1 and earlier)

### ISSUE THREAD

**Issue by paluchasz** [PyGreSQL/PyGreSQL#52]
There seems to be this reference counting bug with some specific queries. The query also has to query a lot of data or be repeated multiple times without restarting script or server. The simplest case I thought off to reproduce seems to be querying for columns of type citext array. For instance can first make this small postgres table

```
CREATE EXTENSION citext;
CREATE TABLE teams (id INT, players CITEXT[]);
INSERT INTO teams VALUES (1, '{"Alice", "Bob"}');
```

and then run the following script

```
import pgdb
import sys


def execute_query(query):
    cursor = CONNECTION.cursor()
    cursor.execute(query)
    rows = cursor.fetchall()
    cursor.close()
    return rows


def reproduce_bug():
    print(sys.getrefcount(None))
    for i in range(10000):
        execute_query('SELECT players FROM teams')
        print(sys.getrefcount(None))


if __name__ == '__main__':
    info = {'user': 'paluchasz', 'host': 'localhost', 'database': 'paluchasz', 'port': 5432, 'password': 'abcd'}
    CONNECTION = pgdb.connect(user=info['user'], host=info['host'], database=info['database'], port=info['port'],
                              password=info['password'])
    reproduce_bug()
```

**Comment by Cito** [comment]
Thank you for reporting this, @paluchasz. I can confirm the problem exists.

**Comment by Cito** [comment]
This is fixed in the master branch now and the fix will be available in version 5.2.2.

**Comment by paluchasz** [comment]
Thanks for resolving! If anyone is looking for a temporary fix then casting the players column to a text array (`players::text[]`) seems to work. I wrote an article about it [here](https://medium.com/python-in-plain-english/resolving-the-fatal-python-error-when-using-pygresql-54c1896929a3)
